helm_express/src/controllers/getPriceHistory.js

```diff
@@ -0,0 +1,55 @@
+const CoinGecko = require("coingecko-api");
+const RedisClient = require("../models/redis.js").client;
+const cache_expire_seconds = require("../models/redis.js").cache_expire_seconds;
+
+async function controller(req, res, next) {
+  /*
+  main logic is as follow:
+  1. check if cache exists
+  2. get price history from CoinGecko
+  3. store result in cache with cache expire seconds
+  */
+  await RedisClient.connect();
+  const tokenName = req.query.tokenName;
+  try {
+    const cacheResult = await RedisClient.get(tokenName);
+    if (cacheResult === null) {
+      // cache doesn't exist!
+      cacheResp = await getPriceHistory(RedisClient, tokenName);
```

Comment:
here's the main logic of `getPriceHistory`

simply fetch data from `CoinGeckoClient`